### DISCUSSION TITLE
How to create list of contacts that are registed in app like WhatsApp

### DISCUSSION THREAD

**Discussion by Niksha36** [GetStream/stream-chat-android#5371] [Q&A]
Guys, I really need help. I know that you can show chats and filter them, but how can you display all users from contacts like in WhatsApp? I have already made an implementation that gets all user contacts and checks whether they are registered or not. The next step is to make an adapter and show these contacts in your recycler view. That is not so easy and i look for a tool in stream sdk to make this right. I also have another idea - just for each registered contact, create a chat with our user and contact where the name of this chat will be the name of the contact, but then what to do with the chat picture? I can't call different pictures for different users for the same chat. Please give me an advise

**Comment by JcMinarro** [comment]
Hello @Niksha36 

It would depend how you register your users.
Stream doesn't require the phone number as field while you register your users.
If you want your users be able to find other users by their phone number, you will need to store it as an extra data for the user.
Once you are able to match your contacts list with the registered users, you could use [Query Users](https://getstream.io/chat/docs/android/query_users/?language=kotlin) endpoint to query for those users.
I would recomend you to implement your own Backend Server to keep this matching process there and be able to have custom queries pasing your contacts list there.

**Reply by Niksha36** [reply]
> Hello @Niksha36 
> 
> It would depend how you register your users.
> Stream doesn't require the phone number as field while you register your users.
> If you want your users be able to find other users by their phone number, you will need to store it as an extra data for the user.
> Once you are able to match your contacts list with the registered users, you could use [Query Users](https://getstream.io/chat/docs/android/query_users/?language=kotlin) endpoint to query for those users.
> I would recomend you to implement your own Backend Server to keep this matching process there and be able to have custom queries pasing your contacts list there.

I use phone number as an id and use firebase auth for verifying number. I did not get your idea: is it a good idea to create chats with all contacts that registed in app. But there is a problem how to show for the same chat another picture and chat name. I want behavior like Whatsapp has. The same chat for two users has user name of a contact and his name

Should I implement my own recycler view ?

**Comment by Niksha36** [comment]
> Hello @Niksha36 
> 
> It would depend how you register your users.
> Stream doesn't require the phone number as field while you register your users.
> If you want your users be able to find other users by their phone number, you will need to store it as an extra data for the user.
> Once you are able to match your contacts list with the registered users, you could use [Query Users](https://getstream.io/chat/docs/android/query_users/?language=kotlin) endpoint to query for those users.
> I would recomend you to implement your own Backend Server to keep this matching process there and be able to have custom queries pasing your contacts list there.

I would like do not close this question

**Comment by aleksandar-apostolov** [comment]
Hi,

You can find our sample app for whats-app clone here:
https://github.com/GetStream/whatsApp-clone-compose

I think it has the functionality you require.
Let us know if this helps you.

**Comment by aleksandar-apostolov** [comment]
Closing this discussion as I believe the sample app provides the necessary guidance and support.
